src/main/java/com/datadoghq/connect/logs/sink/DatadogLogsApiWriter.java

```diff
@@ -100,7 +114,13 @@ private String getOutput(InputStream input) throws IOException {
     }
 
     private void sendRequest(JsonObject content, URL url) throws IOException {
-        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+        HttpURLConnection con;
+        if (!config.proxyURL.isEmpty()) {
+            Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(config.proxyURL, config.proxyPort));
```

Comment:
Not necessary, as the Connector only supports HTTPS anyways (which is the recommended way by Logs/Event Intake)

Reply:
As far as I know you can send HTTPS requests over a SOCKS proxy.

Reply:
I believe there would be issue with our exact configuration (HTTPS with compression). See [this](https://docs.datadoghq.com/agent/logs/log_transport/?tab=https#enforce-a-specific-transport) part of our docs, which says:
```
Note: Setting up a SOCKS5 proxy server enforces TCP transport because socks5 proxies are not yet supported in HTTPS with compression.
```